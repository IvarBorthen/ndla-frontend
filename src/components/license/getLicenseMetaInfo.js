/*
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { metaTypes } from 'ndla-licenses';

function mkCreditsString(group, t, ignoreType) {
  return group
    .map(creator => {
      if (creator.type.toLowerCase() === ignoreType) {
        return creator.name;
      }
      const type = t(`creditType.${creator.type.toLowerCase()}`);
      return `${type} ${creator.name}`;
    })
    .join(', ');
}

export function getLicenseMetaInfo(copyright, t) {
  const creators = mkCreditsString(copyright.creators, t, 'originator');
  const rightsholders = mkCreditsString(
    copyright.rightsholders,
    t,
    'rightsholder',
  );
  const processors = mkCreditsString(copyright.processors, t, 'processor');

  return [
    {
      label: t(`creditType.originator`),
      description: creators,
      metaType: metaTypes.author,
    },
    {
      label: t(`creditType.rightsholder`),
      description: rightsholders,
      metaType: metaTypes.copyrightHolder,
    },
    {
      label: t(`creditType.processor`),
      description: processors,
      metaType: metaTypes.contributor,
    },
  ].filter(item => item.description !== '');
}
