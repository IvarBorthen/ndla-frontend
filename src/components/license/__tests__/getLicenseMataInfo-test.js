/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { getLicenseMetaInfo } from '../getLicenseMetaInfo';

const t = string => {
  const arr = string.split('.');
  return arr[arr.length - 1];
};
test('simple creators array', () => {
  const copyright = {
    creators: [
      {
        type: 'Originator',
        name: 'Siri Knudsen',
      },
    ],
    processors: [],
    rightsholders: [],
  };

  expect(getLicenseMetaInfo(copyright, t)).toMatchSnapshot();
});

test('creators writer and artist', () => {
  const copyright = {
    creators: [
      {
        type: 'Writer',
        name: 'Ernst Hemingway',
      },
      {
        type: 'Artist',
        name: 'Salvador DalÃ­',
      },
    ],
    processors: [],
    rightsholders: [],
  };

  expect(getLicenseMetaInfo(copyright, t)).toMatchSnapshot();
});
test('creators processors and rightsholders', () => {
  const copyright = {
    creators: [
      {
        type: 'Director',
        name: 'Francis Ford Coppola',
      },
    ],
    processors: [
      {
        type: 'Processor',
        name: 'Francis Ford Coppola',
      },
      {
        type: 'Linguistic',
        name: 'Mario Puzo',
      },
    ],
    rightsholders: [
      {
        type: 'Rightsholder',
        name: 'Paramount Pictures',
      },
      {
        type: 'Distributor',
        name: 'Alfran Productions',
      },
    ],
  };

  expect(getLicenseMetaInfo(copyright, t)).toMatchSnapshot();
});
