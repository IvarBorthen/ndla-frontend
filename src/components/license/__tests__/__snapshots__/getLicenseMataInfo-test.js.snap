// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creators processors and rightsholders 1`] = `
Array [
  Object {
    "description": "director Francis Ford Coppola",
    "label": "originator",
    "metaType": "author",
  },
  Object {
    "description": "Paramount Pictures, distributor Alfran Productions",
    "label": "rightsholder",
    "metaType": "copyrightHolder",
  },
  Object {
    "description": "Francis Ford Coppola, linguistic Mario Puzo",
    "label": "processor",
    "metaType": "contributor",
  },
]
`;

exports[`creators writer and artist 1`] = `
Array [
  Object {
    "description": "writer Ernst Hemingway, artist Salvador Dal√≠",
    "label": "originator",
    "metaType": "author",
  },
]
`;

exports[`simple creators array 1`] = `
Array [
  Object {
    "description": "Siri Knudsen",
    "label": "originator",
    "metaType": "author",
  },
]
`;
